### PR TITLE
chore: Move config docs into the config file

### PR DESCRIPTION
**Pull request by kayra1**
# Description

Moves the explanations for the config options into the yaml file itself from https://discourse.charmhub.io/t/reference-filter-configuration/14027

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -3,45 +3,93 @@ options:
     default: True
     type: boolean
     description: |
-      Limit requested identifiers (hostnames, IPs and OIDs) to first requester
+      This filter is disabled by default, and can be enabled by setting the following config:
```
Comment (gruyaume) [inline config.yaml]: "disabled by default" does not seem to match with the default config value.

Comment (gruyaume) [inline config.yaml]: No need to tell users how they can set configs, this is applicable to every config values in the juju ecosystem. This applies at a couple of places in this PR.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 1 above)
Comment (kayra1) [reply config.yaml]: switched the description's places.

Comment (kayra1) [reply config.yaml]: removed all of them

File: config.yaml [hunk 2]
```diff
@@ -3,45 +3,84 @@ options:
     default: True
     type: boolean
     description: |
-      Limit requested identifiers (hostnames, IPs and OIDs) to first requester
+      This filter is enabled by default
```
Comment (gruyaume) [inline config.yaml]: No need to state this, this is automatically rendered:
image

This applies at a couple of places.

File: config.yaml [hunk 3]
```diff
@@ -3,45 +3,84 @@ options:
     default: True
     type: boolean
     description: |
-      Limit requested identifiers (hostnames, IPs and OIDs) to first requester
```
Comment (gruyaume) [inline config.yaml:6]: I think we should keep this as the first sentence of the description. It's a good summary. This is likely true of the other config options as well.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: config.yaml (hunk 2 above)
Comment (kayra1) [reply config.yaml]: removed

File: config.yaml (hunk 3 above)
Comment (kayra1) [reply config.yaml:6]: Replaced the first sentences with the old shorter sentences.

This is harder than I thought :D

**Review by gruyaume — APPROVED** [review]

**Pull request merged**